### PR TITLE
docs: add brief document how to convert home directories to homed

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: docs/CONVERTING_TO_HOMED.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+   Save the information reported by these three commands somewhere, for later
+   reference.
+
+4. Now edit your `/etc/passwd` file and remove your existing record
+   (i.e. delete a single line, the one of your user's account, leaving all
+   other lines unmodified). Similar for `/etc/shadow`, `/etc/group` (in case
+   you have a private group for your user) and `/etc/gshadow`. Most
+   distributions provide you with a tool for that, that adds safe
+   synchronization for these changes: `vipw`, `vipw -s`, `vigr` and `vigr -s`.
```
Comment (keszybz) [inline docs/CONVERTING_TO_HOMED.md:85]: Why not just do `groupdel` and `userdel`? That's generally quicker and safer than editing files.

File: docs/CONVERTING_TO_HOMED.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+   automatically unmounted again. Since we used `--delete-during` all files
+   copied are removed from the old home directory as the copy progresses. After
+   the command completes the old home directory should be empty. Let's remove
+   it hence:
+
+    ```
+    rmdir /home/foobar.saved
+    ```
+
+7. Log out as root again by typing `exit` or similar.
```
Comment (keszybz) [inline docs/CONVERTING_TO_HOMED.md]: Maybe remove this point, and update the para below to say:
> And that's it, we are done already. You can log out now should be able to log in under your user account as usual, but now with `systemd-homed`  managing your home directory.

File: docs/CONVERTING_TO_HOMED.md (hunk 1 above)
Comment (poettering) [reply docs/CONVERTING_TO_HOMED.md:85]: Wel, userdel/groupdel are named differently on the various distros, and I didn't want to spend too much time on that. Also there's just too much opaque stuff behind this commands we don't want. For example, there's a callout script in Fedora's userdel which is supposed to remove at/cron scripts, and I am not sure we should do that. We really just want to remove the record itself, hence I figured telling the user to remove it directly is easiest, clearest and the least likely to have other effects that we do not want.

**Comment by tope324** [comment]
STOP THIS EMAIL NOW.
THIS IS NOT A GITHUB CONTACT EMAIL.

Toby Cockerell

On 9 Apr 2020, at 21:10, Zbigniew Jędrzejewski-Szmek <notifications@github.com> wrote:

﻿

@keszybz commented on this pull request.

Looks good. Some minor suggestions...

________________________________

In docs/CONVERTING_TO_HOMED.md<https://github.com/systemd/systemd/pull/15380#discussion_r406446698>:

> +   This will tell you the name of that group. If the name is the same as your
+   user name your distribution apparently provided you with a private group for
+   your user. If it doesn't match (and is something like `users`) it apparently
+   didn't. Note that `systemd-homed` will always manage a private group for
+   each user under the same name, hence if your distribution is one of the
+   latter kind, then there's a (minor) mismatch in structure when converting.
+
+   Save the information reported by these three commands somewhere, for later
+   reference.
+
+4. Now edit your `/etc/passwd` file and remove your existing record
+   (i.e. delete a single line, the one of your user's account, leaving all
+   other lines unmodified). Similar for `/etc/shadow`, `/etc/group` (in case
+   you have a private group for your user) and `/etc/gshadow`. Most
+   distributions provide you with a tool for that, that adds safe
+   synchronization for these changes: `vipw`, `vipw -s`, `vigr` and `vigr -s`.


Why not just do groupdel and userdel? That's generally quicker and safer than editing files.

________________________________

In docs/CONVERTING_TO_HOMED.md<https://github.com/systemd/systemd/pull/15380#discussion_r406448365>:

> +
+   This mounts the home directory of the user, and then runs the specified
+   `rsync` command which copies the contents of the old home directory into the
+   new. The new home directory is the working directory of the invoked `rsync`
+   process. We are invoking this command as root, hence the `rsync` runs as
+   root too. When the `rsync` command completes the home directory is
+   automatically unmounted again. Since we used `--delete-during` all files
+   copied are removed from the old home directory as the copy progresses. After
+   the command completes the old home directory should be empty. Let's remove
+   it hence:
+
+    ```
+    rmdir /home/foobar.saved
+    ```
+
+7. Log out as root again by typing `exit` or similar.


Maybe remove this point, and update the para below to say:

And that's it, we are done already. You can log out now should be able to log in under your user account as usual, but now with systemd-homed managing your home directory.

**Comment by poettering** [comment]
@tope324 then unregister it from github? we have nothing to do with that.

**Comment by poettering** [comment]
Force pushed a new version. Left the "vipw" stuff in there, instead of uderdel/groupdel, for the aforementioned reasons. Made the other change.

File: docs/CONVERTING_TO_HOMED.md (hunk 1 above)
Comment (keszybz) [reply docs/CONVERTING_TO_HOMED.md:85]: > there's a callout script in Fedora's userdel which is supposed to remove at/cron scripts

Do you have any more information about this? I looked at shadow-utils (which provides the commands) and cronie-* packages, and I didn't see anything relevant. A quick tests with removing and recreating a user shows that user's crontab entries are preserved.

Comment (poettering) [reply docs/CONVERTING_TO_HOMED.md:85]: `man 8 userdel`:

```
       USERDEL_CMD (string)
           If defined, this command is run when removing a user. It should
           remove any at/cron/print jobs etc. owned by the user to be removed
           (passed as the first argument).

           The return code of the script is not taken into account.

           Here is an example script, which removes the user's cron, at and
           print jobs:
```

Note sure Fedora actually makes use of this, but it's definitely something userdel knows and does...

Comment (keszybz) [reply docs/CONVERTING_TO_HOMED.md:85]: I'm pretty sure that this is something that almost nobody knows about. It is (or was, more likely) used on big installations. The number of *laptops*, which is what homed is targetting, using this is probably zero. OTOH, I think the chances that people screw up on of their passwd files when editing them by hand is >0. But let's close this thread.

**Pull request merged**